testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -36,8 +36,11 @@
   begin
     Timeout.timeout(DEFAULT_TIMEOUT) do
       loop do
-        break if page.has_content?(text)
-        sleep 3
+        begin
+          break if page.has_content?(text)
```

Comment:
Instead of retry after 3 seconds, I suggest to increase the timeout

Reply:
That's not the point

Reply:
Similar to that improvement https://github.com/SUSE/spacewalk/pull/7399/files#diff-69eaaf5982104837a8e6521c7fa3d4feR192

Reply:
break if page.has_text?(text, wait: CLICK_TIMEOUT)

Reply:
Ah you want to remove the loop. I thought you wanted to increase the 3 seconds timeout to some other value. Read what you wrote :-) .

@cbosdo , do you think it would solve the page not displayed problem as well ? I think it would.

Reply:
no idea... I hope that would trap the exceptions too

Reply:
Then no loop, just
``` 
raise unless page.has_text?(text, wait: DEFAULT_TIMEOUT)
```

Reply:
We could at least give it a try...

I think it should not generate exceptions as long as it did not finish the DEFAULT_TIMEOUT. And when it reaches it we want an exception anyway.

I hope I'm not believing in Santa Klaus ;-) .

Reply:
OR

we keep that code AND we use it in next step as well (`I wait until I do not see`). Because I don't think that capybara has a `page.has_not_text?`method. The reason for keeping the loop would be homogeneity.

Reply:
I'm not that too keen on relying on assumptions on the internals of capybara right before a release...

Reply:
@Bischoff just pushed according to your last comment

Reply:
OK. We can keep Oscar's idea for another time. It's nice, but I'm not sure how it would work for the negative case where we check the page has *not* a given content.

Reply:
If we don't rely in our test framework... we can turn out lights and go home xD
Yes, Capybara has also the method has_not_text ofc ;)

Reply:
changed

Reply:
ok